packages/mui-material/src/Modal/ModalManager.ts

```diff
@@ -120,45 +149,53 @@ function handleContainer(containerInfo: Container, props: ManagedModalProps) {
         });
         element.style.paddingRight = `${getPaddingRight(element) + scrollbarSize}px`;
       });
-    }
 
-    let scrollContainer: HTMLElement;
+      containerToTest = containerToTest.parentElement;
+    }
 
+    const scrollContainers: HTMLElement[] = [];
     if (container.parentNode instanceof DocumentFragment) {
-      scrollContainer = ownerDocument(container).body;
+      scrollContainers.push(ownerDocument(container).body);
     } else {
       // Support html overflow-y: auto for scroll stability between pages
       // https://css-tricks.com/snippets/css/force-vertical-scrollbar/
-      const parent = container.parentElement;
+      let parent = container.parentElement;
       const containerWindow = ownerWindow(container);
-      scrollContainer =
-        parent?.nodeName === 'HTML' &&
-        containerWindow.getComputedStyle(parent).overflowY === 'scroll'
-          ? parent
-          : container;
+      // We need to find all the scrollable ancestors to
+      // prevent scrolling the modal window
+      while (parent != null) {
+        if (containerWindow.getComputedStyle(parent).overflowY === 'scroll') {
```

Comment:
Why not the HTML check as well from before? (`parent?.nodeName === 'HTML'`)

Reply:
It was for this

```
// Support html overflow-y: auto for scroll stability between pages
      // https://css-tricks.com/snippets/css/force-vertical-scrollbar/
```

I guess it was to support dev who use this css trick, but the thing is `html` can also be scrollable without this explicit trick (see mui demo). Because I changed the code so it goes up the tree in a loop to find the scrollable items naturally `html` element will be included, so no need for that check especially I'm trying to find all scrollable elements.